db.go

```diff
@@ -312,14 +333,33 @@ func (db *DB) retentionCutoff() (b bool, err error) {
 		return false, nil
 	}
 
+	// Size based retention, if MaxBytes is still -1, it has not been set by the user
+	// and only time based retention will be used.
+	var dirsBySize []string
+	if (db.opts.MaxBytes > 0){
```

Comment:
You don't need parentheses around `if` expressions in Go. Same below.

Also, it doesn't seem like all the code has been formatted correctly, could you run `gofmt` (or configure your editor to run it automatically on save)?